crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,236 @@
+//! Types relating to the monitor API.
+
+use crate::Status;
+use serde::{de::Unexpected, Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+use warg_crypto::hash::AnyHash;
+use warg_crypto::signing;
+use warg_protocol::registry::RegistryLen;
+
+/// Represents checkpoint verification response in one of the following states:
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "state", rename_all = "camelCase")]
+#[allow(clippy::large_enum_variant)]
+pub enum CheckpointVerificationResponse {
```

Comment:
Coming back here after reading through the variants and errors below: I wonder if we'd be better off modeling everything other than "verified" as an error here (with `signature_verified: bool` merged into "verified" as suggested below). We could instead categorize errors as either "crticial" (something is positively wrong) or "non-critical" (either "maybe later" or "we don't know"). As-is both kinds of issues are sort of arbitrarily split between OK and error variants.

Reply:
Yeah, thinking about refactoring, as I alluded to in https://github.com/bytecodealliance/registry/pull/225#discussion_r1391489723